lib/pure/rationals.nim

```diff
@@ -40,17 +40,17 @@ func reduce*[T: SomeInteger](x: var Rational[T]) =
     reduce(r)
     doAssert r.num == 1
     doAssert r.den == 2
-
+  if x.den == 0:
+    raise newException(DivByZeroDefect, "division by zero")
```

Comment:
Why doesn't the following `gcd` call do that?

Reply:
i thought it might do that, but the rationale is that
gcd(v,0) = v "because everything divides 0" [wiki](https://en.wikipedia.org/wiki/Binary_GCD_algorithm)